File: koola10/stellar/account.go
package stellar

import (
	"github.com/stellar/go/clients/horizonclient"
	"github.com/stellar/go/keypair"
)

func CreateAccount() (string, string, error) {
	pair, err := keypair.Random()
	if err != nil {
		return "", "", err
	}

	// Fund the account using Friendbot on Testnet
	_, err = horizonclient.DefaultTestNetClient.Fund(pair.Address())
	if err != nil {
		return pair.Address(), pair.Seed(), err
	}

	return pair.Address(), pair.Seed(), nil
}

File: backend/gateway/src/stellar/account.ts
import * as StellarSdk from '@stellar/stellar-sdk';
import axios from 'axios';

export async function createAccount() {
    const pair = StellarSdk.Keypair.random();

    try {
        // Fund the account using Friendbot on Testnet
        const response = await axios.get(`https://friendbot.stellar.org?addr=${pair.publicKey()}`);
        return {
            publicKey: pair.publicKey(),
            secret: pair.secret(),
            response: response.data
        };
    } catch (e) {
        console.error('Error creating/funding account:', e);
        throw e;
    }
}
